Modules/_hashopenssl.c

```diff
@@ -1757,22 +1754,25 @@ _openssl_hash_name_mapper(const EVP_MD *md, const char *from,
 
 
 /* Ask OpenSSL for a list of supported ciphers, filling in a Python set. */
-static PyObject*
-generate_hash_name_list(void)
+static int
+hashlib_md_meth_names(PyObject *module)
 {
     _InternalNameMapperState state;
     state.set = PyFrozenSet_New(NULL);
     if (state.set == NULL)
-        return NULL;
+        return -1;
     state.error = 0;
 
     EVP_MD_do_all(&_openssl_hash_name_mapper, &state);
 
     if (state.error) {
         Py_DECREF(state.set);
-        return NULL;
+        return -1;
+    }
+    if (PyModule_AddObject(module, "openssl_md_meth_names", state.set)) {
+        return -1;
```

Comment:
There should be add `Py_DECREF(state.set)` after `PyModule_AddObject` operation failed?

Reply:
@vstinner pointed this out and I already addressed the problem.